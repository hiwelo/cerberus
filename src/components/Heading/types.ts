import { HTMLAttributes } from 'react';

/** Possible markup overrides for the Heading tag component */
export type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
  /** Specify the tag to use for this heading   */
  tag?: HeadingTag;
}
