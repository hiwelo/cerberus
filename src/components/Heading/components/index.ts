export { default as HeadingTag } from './HeadingTag';
