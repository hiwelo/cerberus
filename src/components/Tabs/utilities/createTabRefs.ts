import { useRef } from 'react';
import { TabsProps } from '../types';

/**
 * Creates a series of React Ref for each panel part of the Tabs defined for
 * this `Tabs` component
 *
 * @param tabs The dictionary of Tabs for the related Tabs component
 */
export function createTabRefs(tabs: TabsProps['tabs']): TabsProps['tabs'] {
  // Transforms the TabListDictionary as an array of tabs
  const tabsEntries = Object.entries(tabs);
  const tabsWithRefs: TabsProps['tabs'] = {};

  // Loops over each tab and creates a React Ref for each panel HTMLElement
  tabsEntries.forEach(([tabKey, tabInfo]) => {
    const refPanel = useRef<HTMLDivElement>(null);

    tabsWithRefs[tabKey] = { ...tabInfo, refPanel };
  });

  return tabsWithRefs;
}
