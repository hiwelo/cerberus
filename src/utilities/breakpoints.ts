/** List of breakpoints in use in our styled components */
export const breakpoints = {
  xs: '0px',
  md: '540px',
};
