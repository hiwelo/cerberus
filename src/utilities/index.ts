export { breakpoints } from './breakpoints';
export { listNodesFromConnection } from './listNodesFromConnection';
