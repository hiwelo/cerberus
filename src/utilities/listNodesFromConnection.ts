/** Describes the generic structure of a GraphQL edge */
type GenericEdge<T> = {
  /** Item from a GraphQL connection object */
  node: T;
  /** Potential typing for the edge added by GraphQL */
  __typename: string;
};

/** Describes the generic structure of a GraphQL connection object */
type GenericConnection<T> = {
  edges: GenericEdge<T>[];
};

/**
 * Returns an array of all nodes contained in the provided GraphQL connection
 * object
 *
 * @param connection GraphQL connection object containing the nodes to return
 */
export function listNodesFromConnection<T>(
  connection?: GenericConnection<T>,
): T[] | undefined {
  return connection?.edges.map(({ node }) => node);
}
