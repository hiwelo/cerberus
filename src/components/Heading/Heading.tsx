import React, { FunctionComponent } from 'react';
import { HeadingTag } from './components';
import { HeadingProps } from './types';

export const Heading: FunctionComponent<HeadingProps> = ({ children, tag }) => {
  return <HeadingTag as={tag}>{children}</HeadingTag>;
};
