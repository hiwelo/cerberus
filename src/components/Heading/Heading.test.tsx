import React from 'react';
import { testA11y, testSnapshot, mountWithApp } from 'utilities/test';
import { Heading } from './Heading';

const TestContainer = <Heading>Test</Heading>;

describe('Heading', () => {
  it('should use the tag provided by the consumer', () => {
    const requestedTag = 'h2';
    const wrapper = mountWithApp(<Heading tag={requestedTag}>Test</Heading>);
    const result = wrapper.getDOMNode();

    expect(result.tagName).toEqual(requestedTag.toUpperCase());
  });

  testSnapshot(TestContainer);
  testA11y(TestContainer);
});
