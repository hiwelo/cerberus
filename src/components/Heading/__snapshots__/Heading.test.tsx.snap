// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match snapshot 1`] = `
.c0 {
  font-size: 2rem;
  font-weight: 600;
  -webkit-letter-spacing: -0.035em;
  -moz-letter-spacing: -0.035em;
  -ms-letter-spacing: -0.035em;
  letter-spacing: -0.035em;
  text-align: left;
}

<h1
  class="c0"
>
  Test
</h1>
`;
